subxt/src/client/lightclient/background.rs

```diff
@@ -0,0 +1,428 @@
+                        .is_err()
+                    {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                } else if let Some((sub_id_sender, _)) = self.id_to_subscription.remove(&id) {
+                    if sub_id_sender
+                        .send(Err(LightClientError::Request(error.to_string())))
+                        .is_err()
+                    {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                }
+            }
+            Ok(RpcResponse::Method { id, result }) => {
+                // Send the response back.
+                if let Some(sender) = self.requests.remove(&id) {
+                    if sender.send(Ok(result)).is_err() {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                } else if let Some((sub_id_sender, sender)) = self.id_to_subscription.remove(&id) {
+                    let sub_id = result
+                        .get()
+                        .trim_start_matches('"')
+                        .trim_end_matches('"')
+                        .to_string();
+                    tracing::trace!(target: LOG_TARGET, "Received subscription ID: {}", sub_id);
+
+                    if sub_id_sender.send(Ok(result)).is_err() {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    } else {
+                        // Track this subscription ID if send is successful.
+                        self.subscriptions.insert(sub_id, sender);
+                    }
+                }
+            }
+            Ok(RpcResponse::Subscription { method, id, result }) => {
+                // Subxt calls into `author_submitAndWatchExtrinsic`, however the smoldot produces
+                // `{"event":"broadcasted","numPeers":1}` and `{"event":"validated"}` which are part
+                // of the RPC V2 API. Ignore those events.
+                if method == "transaction_unstable_watchEvent"
+                    && (result.to_string().contains("broadcasted")
+                        || result.to_string().contains("validated"))
+                {
+                    tracing::debug!(target: LOG_TARGET, "Ignoring notification {:?}", result);
+                    return;
+                }
```

Comment:
Same remark here. This is fixed.

Reply:
Yep, it totally works now! Thanks a lot for fixing this and for the review! 🙏